WooCommerce/Classes/Tools/SiteCountry.swift

```diff
@@ -0,0 +1,664 @@
+import Foundation
+import Yosemite
+
+final class SiteCountry {
+    /// ResultsController. Fetches the store country from SiteSetting
+    ///
+    private lazy var resultsController: ResultsController<StorageSiteSetting> = {
+        let storageManager = AppDelegate.shared.storageManager
+        let sitePredicate = NSPredicate(format: "siteID == %lld", StoresManager.shared.sessionManager.defaultStoreID ?? Int.min)
+        let settingCountryPredicate = NSPredicate(format: "settingID ==[c] %@", Constants.countryKey)
+
+        let compoundPredicate = NSCompoundPredicate(andPredicateWithSubpredicates: [sitePredicate, settingCountryPredicate])
+
+        let siteIDKeyPath = #keyPath(StorageSiteSetting.siteID)
+        let descriptor = NSSortDescriptor(keyPath: \StorageSiteSetting.siteID, ascending: false)
+        return ResultsController<StorageSiteSetting>(storageManager: storageManager,
+                                                     sectionNameKeyPath: siteIDKeyPath,
+                                                     matching: compoundPredicate,
+                                                     sortedBy: [descriptor])
+    }()
+
+    init() {
+        configureResultsController()
+    }
+
+    private var siteCountry: String? {
+        return resultsController.fetchedObjects.first?.value
+    }
+
+    /// Returns the name of the country associated with the current store.
+    /// The default store country is provided in a format like `HK:KOWLOON`
+    /// This methdo will transform `HK:KOWLOON` into `Hong Kong`
+    /// Will return nil if it can not figure out a valid country name
+    var siteCountryName: String? {
+        guard let siteCountryCode = siteCountry,
+            let code = siteCountryCode.components(separatedBy: ":").first,
+            let countryCode = CountryCode(rawValue: code) else {
+                return nil
+        }
+
+        return countryCode.readableCountry
+    }
+
+    /// Setup: ResultsController
+    ///
+    private func configureResultsController() {
+        try? resultsController.performFetch()
+    }
+}
+
+
+// MARK: - Mapping between country codes and readable names
```

Comment:
It would be good to link to the source for the country codes list. Did you find this in the WC API docs?